src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/Extensions/SAP_Internal_Stats_KeyFigure+Formatting.swift

```diff
@@ -19,13 +19,28 @@ extension SAP_Internal_Stats_KeyFigure {
 				return nil
 			}
 
-			return String(format: AppStrings.Statistics.Card.million, formattedNumber)
+			formattedString = String(format: AppStrings.Statistics.Card.million, formattedNumber)
 		} else {
 			let decimals = max(0, Int(self.decimals))
 			numberFormatter.minimumFractionDigits = Int(decimals)
 			numberFormatter.maximumFractionDigits = Int(decimals)
 
-			return numberFormatter.string(from: NSNumber(value: value))
+			formattedString = numberFormatter.string(from: NSNumber(value: value))
+		}
+		guard let cardId = cardRawValue else {
+			return formattedString
+		}
+		switch HomeStatisticsCard(rawValue: cardId) {
+		case .atLeastOneVaccinatedPerson, .fullyVaccinatedPeople:
+			guard let valueString = formattedString else {
+				return formattedString
+			}
+			return valueString + AppStrings.Statistics.percent
+		case .infections, .incidence, .keySubmissions, .reproductionNumber, .appliedVaccinationsDoseRates:
+			return formattedString
+		case .none:
+			Log.info("Statistics card ID \(cardId) is not supported", log: .ui)
+			return formattedString
```

Comment:
this is basically the same case as line 30 & 31
better call the method `formattedValue` with parameter `HomeStatisticsCard` so you can remove line 30 & 31 checks?